cell_based/src/population/ImmersedBoundaryCellPopulation.hpp

```diff
@@ -0,0 +1,679 @@
+
+    /**
+     * Overridden GetNeighbouringLocationIndices() method. Given a cell, returns 
+     * the set of location indices corresponding to neighbouring cells. At least 
+     * once per timestep, UpdateCellCentroids() should be called before this 
+     * function.
+     *
+     * @param pCell a cell
+     * 
+     * @return the set of neighbouring location indices.
+     */
+    std::set<unsigned> GetNeighbouringLocationIndices(CellPtr pCell);
+
+    /**
+     * Overridden AddNode() method. Add a new node to the cell population.
+     *
+     * @param pNewNode pointer to the new node
+     * 
+     * @return global index of new node in cell population
+     */
+    unsigned AddNode(Node<DIM>* pNewNode);
+
+    /**
+     * Overridden UpdateNodeLocations() method.
+     *
+     * @param dt the time step
+     */
+    void UpdateNodeLocations(double dt);
+
+    /**
+     * Overridden SetNode() method. Move the node with a given index to a new 
+     * point in space.
+     *
+     * @param index the index of the node to be moved
+     * 
+     * @param rNewLocation the new target location of the node
+     */
+    void SetNode(unsigned index, ChastePoint<DIM>& rNewLocation);
+
+    /**
+     * Get a pointer to the element corresponding to a given CellPtr.
+     *
+     * @param pCell the cell
+     *
+     * @return pointer to the element.
+     */
+    ImmersedBoundaryElement<DIM, DIM>* GetElementCorrespondingToCell(
+        CellPtr pCell);
+
+    /**
+     * Overridden AddCell() method.
+     *
+     * Add a new cell to the cell population.
+     *
+     * @param pNewCell  the cell to add
+     * @param pParentCell pointer to a parent cell (if required)
+     * 
+     * @return address of cell as it appears in the cell list (internal of this 
+     *     method uses a copy constructor along the way)
+     */
+    CellPtr AddCell(CellPtr pNewCell, CellPtr pParentCell = CellPtr());
+
+    /**
+     * Remove all cells labelled as dead. Note that after calling this method 
+     * the cell population will be in an inconsistent state until the equivalent 
+     * of a 'remesh' is performed! So don't try iterating over cells or anything
+     * like that.
+     *
+     * @return number of cells removed
+     */
+    unsigned RemoveDeadCells();
+
+    /**
+     * Overridden IsCellAssociatedWithADeletedLocation() method.
+     *
+     * @param pCell the cell
+     * @return whether a given cell is associated with a deleted element.
+     */
+    bool IsCellAssociatedWithADeletedLocation(CellPtr pCell);
+
+    /**
+     * Remove the ImmersedBoundaryElements which have been marked as deleted, 
+     * perform any cell rearrangements if required, and update the 
+     * correspondence with CellPtrs.
+     *
+     * @param hasHadBirthsOrDeaths - a bool saying whether cell population has 
+     *     had Births Or Deaths not needed in this cell population class
+     * 
+     * \todo why not needed? Cells should be able to divide and/or die in 
```

Comment:
lurking todo ?

Reply:
It's not really resolved if the code still has a `\todo` comment in it though? I can live with a new issue being opened for these and pressing ahead with the release!

Reply:
The comment has been removed now. Interestingly I don't think that parameter is used once in the entire chaste codebase so it's definitely worth an issue!